src/tests/test_types.cpp

```diff
@@ -264,4 +264,72 @@ INSTANTIATE_TEST_SUITE_P(TestStringSuite, TestString,
                                            TestStringParam("abc", ""),
                                            TestStringParam("bb", "aa")));
 
-}  // namespace tests
+struct TestDateTypeParam {
+    std::string const l;
+    std::string const r;
+
+    TestDateTypeParam(std::string l, std::string r) noexcept : l(std::move(l)), r(std::move(r)) {}
+};
+
+class TestDateType : public ::testing::TestWithParam<TestDateTypeParam> {};
+
+TEST_P(TestDateType, ComparisonTest) {
+    TestDateTypeParam p = GetParam();
+    auto const& [l_val, r_val] = GetParam();
+    std::unique_ptr<mo::Type> type(mo::CreateType(mo::TypeId::kDate, true));
+    auto* date_type = static_cast<mo::DateType*>(type.get());
```

Comment:
Why not `std::unique_ptr<mo::Type> date_type(mo::CreateSpecificType<mo::DateType>(mo::TypeId::kDate, true));`?

Reply:
The same problem in the test below